File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m=0;
        int n=prices.size();
        int c=prices[0];
        for(int i=1;i<n;i++){
            if(prices[i]>c)
                m=max(m,prices[i]-c);
            else
                c=prices[i];
        }
        return m;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let result = 0,
		min = Infinity;
	prices.forEach((price) => {
		if (price < min) min = price;
		else if (price - min > result) result = price - min;
	});
	return result;
};